src/model/CodacyInput.ts

```diff
@@ -0,0 +1,41 @@
+export type ParameterValue = any
+
+export class Parameter {
+  readonly name: string
+  readonly value: ParameterValue
+
+  constructor(name: string, value: ParameterValue) {
+    this.name = name
+    this.value = value
+  }
+}
+
+export class Pattern {
+  readonly patternId: string
+  readonly parameters?: Parameter[]
+
+  constructor(patternId: string, parameters?: Parameter[]) {
```

Comment:
arrays should degraded to empty array not to `undefined`

Reply:
everywhere in this file

Reply:
I used these types because it matches the structure of the files we use in Codacy.
For example, a valid Pattern in our .codacyrc file can have no parameters section at all.
So I can parse the Json file and have a Pattern object directly.
Otherwise I should define a structure like this for parsing, and another structure with no undefined fields and convert by hand to it after parsing.